Add Flock 2025 workshop content (#1050)
packit/packit.dev: src/pages/flock-2025-workshop.md

# Smarter package maintenance with Packit (Flock 2025)

## 🎯 Goal

The main goal of this workshop is to **set up a complete Fedora release pipeline** for your package using the following Packit jobs:

- `pull_from_upstream` (alternatively `propose_downstream`, if you already use Packit in upstream)
- `koji_build`
- `bodhi_update`

By the end of this session, you should have a working configuration that automates these steps in the Fedora package release lifecycle.

---

## 🛠️ Instructions

### Prerequisites

> - Ensure you have a **Fedora package** you maintain/co-maintain. Ideally, pick one that has a **pending upstream release**, as this allows trying things out easily.

Reviewer: ```suggestion
Ensure you have a **Fedora package** you maintain/co-maintain. Ideally, pick one that has a **pending upstream release**, as this allows trying things out easily.
```

### Setup

> For detailed setup instructions, you can follow [the documentation](https://packit.dev/docs/fedora-releases-guide/dist-git-onboarding#setup). You can find a brief summary below:

Reviewer 2: ```suggestion
For detailed setup instructions, you can follow [the documentation](https://packit.dev/docs/fedora-releases-guide/dist-git-onboarding#setup). You can find a brief summary below.
```

> #### Create a working branch

Reviewer 2: Maybe mention also a fork to not push the branch to the `rpms/whatever`? You can push, you cannot delete afterwards, I already have one branch like that :/

Before creating the configuration file, review [this section on how you'll be trying out the setup](https://packit.dev/docs/fedora-releases-guide/dist-git-onboarding#release-syncing).
Based on that, create a new branch in your dist-git repository for your Packit configuration
(e.g., `packit-setup`).

#### Create the Packit configuration file

Packit uses a YAML configuration file, named `(.)packit.yaml`, to define its behavior. This file needs to be placed in your **dist-git repository**,

> in general in the `rawhide` branch, but for experimentation during the workshop in a new branch, trying things out in a pull request.

Reviewer 2: ```suggestion
in general in the `rawhide` branch, but for experimentation during the workshop in a new branch, to be able to try running the jobs from an open pull request.
```

Then, the simplest ways of creating the config:

> - Use `packit dist-git init` (requires `packit` installed locally, e.g. `dnf install packit`)

Reviewer 2: ```suggestion
- Use `packit dist-git init` (requires `packit` installed locally, e.g., `dnf install packit`)
```

Oxford comma above, for consistency
